diff --git a/Android.mk b/Android.mk
index 7fd658b..bb04e14 100644
--- a/Android.mk
+++ b/Android.mk
@@ -95,4 +95,7 @@ endif
 ifeq ($(TARGET_PRODUCT), evk_8mq)
     include $(LOCAL_PATH)/conf/fsl/evk_8mq/Android.mk
 endif
+ifeq ($(TARGET_PRODUCT), visionsom_8mm)
+    include $(LOCAL_PATH)/conf/fsl/visionsom_8mm/Android.mk
+endif
 endif # BOARD_HAVE_BLUETOOTH_BCM
diff --git a/conf/fsl/visionsom_8mm/Android.mk b/conf/fsl/visionsom_8mm/Android.mk
new file mode 100644
index 0000000..49cb335
--- /dev/null
+++ b/conf/fsl/visionsom_8mm/Android.mk
@@ -0,0 +1,10 @@
+LOCAL_PATH := $(call my-dir)
+
+include $(CLEAR_VARS)
+
+LOCAL_MODULE := bt_vendor.conf
+LOCAL_MODULE_CLASS := ETC
+LOCAL_MODULE_PATH := $(TARGET_OUT_VENDOR)/etc/bluetooth
+LOCAL_SRC_FILES := $(LOCAL_MODULE)
+
+include $(BUILD_PREBUILT)
diff --git a/conf/fsl/visionsom_8mm/bt_vendor.conf b/conf/fsl/visionsom_8mm/bt_vendor.conf
new file mode 100644
index 0000000..7448981
--- /dev/null
+++ b/conf/fsl/visionsom_8mm/bt_vendor.conf
@@ -0,0 +1,6 @@
+# UART device port where Bluetooth controller is attached
+UartPort = /dev/ttymxc0
+
+# Firmware patch file location
+FwPatchFilePath = /vendor/firmware/brcm/
+FwPatchFileName = BCM43430A1.hcd
diff --git a/include/bt_vendor_brcm.h b/include/bt_vendor_brcm.h
index 545e76e..03dea5b 100644
--- a/include/bt_vendor_brcm.h
+++ b/include/bt_vendor_brcm.h
@@ -51,12 +51,12 @@
 
 /* Run-time configuration file */
 #ifndef VENDOR_LIB_CONF_FILE
-#define VENDOR_LIB_CONF_FILE "/etc/bluetooth/bt_vendor.conf"
+#define VENDOR_LIB_CONF_FILE "/vendor/etc/bluetooth/bt_vendor.conf"
 #endif
 
 /* Device port name where Bluetooth controller attached */
 #ifndef BLUETOOTH_UART_DEVICE_PORT
-#define BLUETOOTH_UART_DEVICE_PORT      "/dev/ttyO1"    /* maguro */
+#define BLUETOOTH_UART_DEVICE_PORT      "/dev/ttymxc0"    /* maguro */
 #endif
 
 /* Location of firmware patch files */
diff --git a/include/vnd_visionsom_8mm.txt b/include/vnd_visionsom_8mm.txt
new file mode 100644
index 0000000..1e46fe0
--- /dev/null
+++ b/include/vnd_visionsom_8mm.txt
@@ -0,0 +1,11 @@
+BLUETOOTH_UART_DEVICE_PORT = "/dev/ttymxc0"
+FW_PATCHFILE_LOCATION = "/vendor/firmware/brcm"
+VENDOR_LIB_CONF_FILE = "/vendor/etc/bluetooth/bt_vendor.conf"
+LPM_IDLE_TIMEOUT_MULTIPLE = 5
+SCO_USE_I2S_INTERFACE = FALSE
+BTVND_DBG = TRUE
+BTHW_DBG = TRUE
+VNDUSERIAL_DBG = TRUE
+UPIO_DBG = TRUE
+LPM_SLEEP_MODE = 0
+UART_TARGET_BAUD_RATE = 1000000
